### PR TITLE
New types conversion [7795]

### PR DESCRIPTION
**Pull request by LuisGP**
This PR adds support for the conversion of the `map`, `union` types, and resolves the `alias` type.

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/soss-dds-linux/30/)

File: packages/dds/src/Conversion.cpp [hunk 1]
```diff
@@ -314,6 +345,15 @@ void Conversion::set_sequence_data(
...
             }
+            case ::xtypes::TypeKind::MAP_TYPE:
+            {
+                DynamicTypeBuilder_ptr builder = get_builder(from[idx].type()); // The inner map builder
+                DynamicTypeBuilder* builder_ptr = static_cast<DynamicTypeBuilder*>(builder.get());
+                DynamicData* seq_data = factory->create_data(builder_ptr->build());
+                set_map_data(from[idx], seq_data);
+                to->set_complex_value(seq_data, id);
+                break;
+            }
```
Comment (jamoralp) [inline packages/dds/src/Conversion.cpp:356]: `xtypes::TypeKind::UNION_TYPE` case missing

Comment (richiware) [inline packages/dds/src/Conversion.cpp:356]: And for UNION_TYPE?

**Review by richiware — DISMISSED** [review]

**Review by jamoralp — CHANGES_REQUESTED** [review]

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/soss-dds-linux/31/)

**Review by jamoralp — APPROVED** [review]
LGTM

**Pull request merged**